From 8ccde114ad0d5c8607b190adfb77b5d2a6b2af03 Mon Sep 17 00:00:00 2001
From: Yi Xie <imxieyi@users.noreply.github.com>
Date: Tue, 25 Mar 2025 23:36:11 +0900
Subject: [PATCH] Add libaribcaption

---
 extras/package/apple/build.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extras/package/apple/build.conf b/extras/package/apple/build.conf
index 767e6fe625..00a50ae170 100644
--- a/extras/package/apple/build.conf
+++ b/extras/package/apple/build.conf
@@ -66,6 +66,7 @@ export VLC_CONTRIB_OPTIONS_BASE=(
     --disable-luac
     --disable-aribb24
     --disable-aribb25
+    --enable-libaribcaption
     --enable-vpx
     --enable-libdsm
     --enable-libplacebo
-- 
2.39.5 (Apple Git-154)

