From 75a0c18fff49e85dfc5fa9a6900b86790eabf268 Mon Sep 17 00:00:00 2001
From: Yi Xie <imxieyi@users.noreply.github.com>
Date: Tue, 25 Mar 2025 23:36:11 +0900
Subject: [PATCH 13/18] Add libaribcaption

---
 extras/package/apple/build.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extras/package/apple/build.conf b/extras/package/apple/build.conf
index 86af8564ad..2ecd302bea 100644
--- a/extras/package/apple/build.conf
+++ b/extras/package/apple/build.conf
@@ -65,6 +65,7 @@ export VLC_CONTRIB_OPTIONS_BASE=(
     --disable-luac
     --disable-aribb24
     --disable-aribb25
+    --enable-libaribcaption
     --enable-vpx
     --enable-libdsm
     --enable-libplacebo
-- 
2.39.5 (Apple Git-154)

