From 91396089543b438a0afd6ba808d5bf5081dd911f Mon Sep 17 00:00:00 2001
From: Yi Xie <imxieyi@users.noreply.github.com>
Date: Wed, 16 Apr 2025 00:34:38 +0900
Subject: [PATCH 18/18] Remove ytdl on macOS

---
 extras/package/apple/build.conf | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/extras/package/apple/build.conf b/extras/package/apple/build.conf
index 2ecd302bea..2f53458729 100644
--- a/extras/package/apple/build.conf
+++ b/extras/package/apple/build.conf
@@ -254,7 +254,9 @@ export VLC_MODULE_REMOVAL_LIST_BASE=(
 )
 
 # Additional modules to remove for macOS
-export VLC_MODULE_REMOVAL_LIST_MACOSX=()
+export VLC_MODULE_REMOVAL_LIST_MACOSX=(
+    ytdl
+)
 
 # Additional modules to remove for iOS
 export VLC_MODULE_REMOVAL_LIST_IOS=()
-- 
2.39.5 (Apple Git-154)

